service/controller/app/resource/catalog/resource.go

```diff
@@ -64,17 +64,36 @@ func (r *Resource) getCatalogForApp(ctx context.Context, customResource v1alpha1
 
 	catalogName := key.CatalogName(customResource)
 
-	r.logger.Debugf(ctx, "looking for appCatalog %#q", catalogName)
+	r.logger.Debugf(ctx, "looking for catalog %#q", catalogName)
 
-	appCatalog, err := r.g8sClient.ApplicationV1alpha1().AppCatalogs().Get(ctx, catalogName, metav1.GetOptions{})
-	if apierrors.IsNotFound(err) {
-		return microerror.Maskf(notFoundError, "appCatalog %#q", catalogName)
-	} else if err != nil {
-		return microerror.Mask(err)
+	var namespaces []string
+	{
+		if customResource.Spec.CatalogNamespace != "" {
+			namespaces = []string{customResource.Spec.CatalogNamespace}
+		} else {
+			namespaces = []string{metav1.NamespaceDefault, "giantswarm"}
+		}
+	}
+
+	var catalog *v1alpha1.Catalog
+	var namespace string
+	for _, namespace := range namespaces {
+		catalog, err = r.g8sClient.ApplicationV1alpha1().Catalogs(namespace).Get(ctx, catalogName, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			// no-op
+			continue
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+		break
+	}
+
+	if catalog == nil {
+		return microerror.Maskf(notFoundError, "catalog %#q", catalogName)
 	}
 
-	r.logger.Debugf(ctx, "found appCatalog %#q", catalogName)
-	cc.AppCatalog = *appCatalog
+	r.logger.Debugf(ctx, "found catalog %#q in namespace %#q", catalogName, namespace)
```

Comment:
I think we should log the namespace from now.

Reply:
Yes, nice. 👍